test/unit/Container.spec.ts

```diff
@@ -128,13 +112,13 @@ describe("Container", function() {
             }
         }
 
-        function createCar (engine: Engine) {
-            return new Car(engine);
+        class CarFactory {
+            @Factory()
+            public static createCar (engine: Engine): Car {
```

Comment:
> we can't decide for the user.

we can't provide everything user wants as well. Lets skip third case for now. If there be many people requesting this feature we will start to think how to implement it the best way. But I doubt many people need such complex functionality.